fe/fe-core/src/main/java/com/starrocks/external/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.io.SeekableInputStream;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Implementation of FileIO that adds metadata content caching features.
+ */
+public class IcebergCachingFileIO implements FileIO {
+    private static final Logger LOG = LogManager.getLogger(IcebergCachingFileIO.class);
+    private static final int BUFFER_CHUNK_SIZE = 4 * 1024 * 1024; // 4MB
+    private static final long DEFAULT_FILEIO_CACHE_MAX_CONTENT_LENGTH = 8 * 1024 * 1024;
+    private static final long DEFAULT_FILEIO_CACHE_MAX_TOTAL_BYTES = 128 * 1024 * 1024;
+
+    public static final String FILEIO_CACHE_MAX_TOTAL_BYTES = "fileIO.cache.max-total-bytes";
+
+    private ContentCache fileContentCache;
+    private FileIO wrappedIO;
+
+    public IcebergCachingFileIO(FileIO io) {
+        this.wrappedIO = io;
+    }
+
+    @Override
+    public void initialize(Map<String, String> properties) {
+        long maxTotalBytes = PropertyUtil.propertyAsLong(properties, FILEIO_CACHE_MAX_TOTAL_BYTES,
+                                                        DEFAULT_FILEIO_CACHE_MAX_TOTAL_BYTES);
+        this.fileContentCache = new ContentCache(DEFAULT_FILEIO_CACHE_MAX_CONTENT_LENGTH, maxTotalBytes);
+    }
+
+    @Override
+    public InputFile newInputFile(String path) {
+        return new CachingInputFile(fileContentCache, wrappedIO.newInputFile(path));
+    }
+
+    @Override
+    public OutputFile newOutputFile(String path) {
+        return wrappedIO.newOutputFile(path);
+    }
+
+    @Override
+    public void deleteFile(String path) {
+        wrappedIO.deleteFile(path);
+        // remove from cache.
+        fileContentCache.invalidate(path);
+    }
+
+    private static class CacheEntry {
+        private final long length;
+        private final List<ByteBuffer> buffers;
+
+        private CacheEntry(long length, List<ByteBuffer> buffers) {
+            this.length = length;
+            this.buffers = buffers;
+        }
+    }
+
+    public static class ContentCache {
+        private final long maxTotalBytes;
+        private final long maxContentLength;
+        private final Cache<String, CacheEntry> cache;
+
+        private ContentCache(long maxContentLength, long maxTotalBytes) {
+            this.maxTotalBytes = maxTotalBytes;
+            this.maxContentLength = maxContentLength;
+
+            Caffeine<Object, Object> builder = Caffeine.newBuilder();
+            this.cache = builder.maximumWeight(maxTotalBytes)
```

Comment:
we may get outdated file if we never touch the maxTotalBytes, am i right?

Reply:
this cache the iceberg meta file,    if  delete  record,  add record, or update in iceberg, will  add new meta file in iceberg